From e1a2075aba12479cef7652247a7384edb1a74af3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Fri, 5 Jan 2024 16:07:57 +0100
Subject: [PATCH 0769/1141] crypto: stm32/cryp - don't return 0 when dma_map_sg
 fail
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When dma_map_sg fails it returns 0.
Then err must not be returned but -ENOMEM instead.

Change-Id: I00043aaa32389cca07d6deeed46283edebe0ff22
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/349590
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Jerome BESNARD <jerome.besnard@st.com>
---
 drivers/crypto/stm32/stm32-cryp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/stm32/stm32-cryp.c b/drivers/crypto/stm32/stm32-cryp.c
index 4e88e522405c..283ec1469b6b 100644
--- a/drivers/crypto/stm32/stm32-cryp.c
+++ b/drivers/crypto/stm32/stm32-cryp.c
@@ -819,7 +819,7 @@ static int stm32_cryp_header_dma_start(struct stm32_cryp *cryp)
 	err = dma_map_sg(cryp->dev, cryp->header_sg, cryp->header_sg_len, DMA_TO_DEVICE);
 	if (!err) {
 		dev_err(cryp->dev, "dma_map_sg() error\n");
-		return err;
+		return -ENOMEM;
 	}
 
 	dma_sync_sg_for_device(cryp->dev, cryp->header_sg, cryp->header_sg_len, DMA_TO_DEVICE);
@@ -863,14 +863,14 @@ static int stm32_cryp_dma_start(struct stm32_cryp *cryp)
 		err = dma_map_sg(cryp->dev, cryp->in_sg, cryp->in_sg_len, DMA_TO_DEVICE);
 		if (!err) {
 			dev_err(cryp->dev, "dma_map_sg() error\n");
-			return err;
+			return -ENOMEM;
 		}
 	}
 
 	err = dma_map_sg(cryp->dev, cryp->out_sg, cryp->out_sg_len, DMA_FROM_DEVICE);
 	if (!err) {
 		dev_err(cryp->dev, "dma_map_sg() error\n");
-		return err;
+		return -ENOMEM;
 	}
 
 	dma_sync_sg_for_device(cryp->dev, cryp->in_sg, cryp->in_sg_len, DMA_TO_DEVICE);
-- 
2.39.2

